src/components/LayoutDialog.vue

```diff
@@ -80,22 +83,71 @@ function handleImageUpload(event: Event) {
   if (input.files && input.files[0]) {
     const file = input.files[0];
     const reader = new FileReader();
-    
+
     reader.onload = (e) => {
       if (e.target?.result) {
-        layout.backgroundImage = {
-          image: e.target.result as string,
-          x: layout.backgroundImage?.x || 0,
-          y: layout.backgroundImage?.y || 0,
-          width: layout.backgroundImage?.width || 10,
-          height: layout.backgroundImage?.height || 10
+        const map_image = new Image();
+        map_image.onload = async () => {
+          // get real width and height
+          const imageWidth = map_image.naturalWidth;
+          const imageHeight = map_image.naturalHeight;
+          // populate layout background image
+          layout.backgroundImage = {
+            image: map_image.src,
+            x: layout.backgroundImage?.x || 0,
+            y: layout.backgroundImage?.y || 0,
+            width: layout.backgroundImage?.width || imageWidth,
+            height: layout.backgroundImage?.height || imageHeight,
+            natural_width: imageWidth,
+            natural_height: imageHeight
+          };
         };
+        map_image.src = e.target.result as string;
+
+        // TODO: if there is a metadata file in the same folder, it should be loaded it here
+        // load_map_metadata(metadata_file);
       }
     };
-    
     reader.readAsDataURL(file);
   }
 }
+
+function handleMetadataUpload(event: Event) {
+  const input = event.target as HTMLInputElement;
+  if (input.files && input.files[0]) {
+    const metadata_file = input.files[0];
+    load_map_metadata(metadata_file);
+  };
+}
+
+function load_map_metadata(metadata_file: File) {
```

Comment:
function and variable names are camelCase

Reply:
also you should check `src/controllers/layout.controller.ts` maybe you can move the functions to there